labelbox/data/annotation_types/data/tiled_image.py

```diff
@@ -0,0 +1,482 @@
+        bounds = values.get('bounds')
+
+        if epsg != EPSG.SIMPLEPIXEL:
+            for bound in bounds:
+                lat, lng = bound.y, bound.x
+                if int(lng) not in VALID_LNG_RANGE or int(
+                        lat) not in VALID_LAT_RANGE:
+                    raise ValueError(f"Invalid lat/lng bounds. Found {bounds}. "
+                                     f"lat must be in {VALID_LAT_RANGE}. "
+                                     f"lng must be in {VALID_LNG_RANGE}.")
+        return values
+
+
+class TileLayer(BaseModel):
+    """ Url that contains the tile layer. Must be in the format:
+
+    https://c.tile.openstreetmap.org/{z}/{x}/{y}.png
+
+    >>> layer = TileLayer(
+        url="https://c.tile.openstreetmap.org/{z}/{x}/{y}.png",
+        name="slippy map tile"
+        )
+    """
+    url: str
+    name: Optional[str] = "default"
+
+    @validator('url')
+    def validate_url(cls, url):
+        xyz_format = "/{z}/{x}/{y}"
+        if xyz_format not in url:
+            raise ValueError(f"{url} needs to contain {xyz_format}")
+        return url
+
+
+class TiledImageData(BaseData):
+    """ Represents tiled imagery 
+
+    If specified version is 2, converts bounds from [lng,lat] to [lat,lng]
+
+    Requires the following args:
+        tile_layer: TileLayer
+        tile_bounds: TiledBounds
+        zoom_levels: List[int]
+    Optional args:
+        max_native_zoom: int = None
+        tile_size: Optional[int]
+        version: int = 2
+        alternative_layers: List[TileLayer]     
+
+    >>> tiled_image_data = TiledImageData(tile_layer=TileLayer,
+                                  tile_bounds=TiledBounds,
+                                  zoom_levels=[1, 12])
+    """
+    tile_layer: TileLayer
+    tile_bounds: TiledBounds
+    alternative_layers: List[TileLayer] = None
+    zoom_levels: Tuple[int, int]
+    max_native_zoom: Optional[int] = None
+    tile_size: Optional[int] = DEFAULT_TMS_TILE_SIZE
+    version: Optional[int] = 2
+    multithread: bool = True
+
+    def as_raster_data(self,
+                       zoom: int = 0,
+                       max_tiles: int = 32,
+                       multithread=True) -> RasterData:
+        """Converts the tiled image asset into a RasterData object containing an
+        np.ndarray.
+
+        Uses the minimum zoom provided to render the image.
+        """
+        if self.tile_bounds.epsg == EPSG.SIMPLEPIXEL:
+            xstart, ystart, xend, yend = self._get_simple_image_params(zoom)
+
+        # Currently our editor doesn't support anything other than 3857.
+        # Since the user provided projection is ignored by the editor
+        #    we will ignore it here and assume that the projection is 3857.
+        elif self.tile_bounds.epsg == EPSG.EPSG3857:
+            xstart, ystart, xend, yend = self._get_3857_image_params(zoom)
+        else:
+            raise ValueError(
+                f"Unsupported epsg found...{self.tile_bounds.epsg}")
```

Comment:
I would get rid of this comment. We aren't ignoring the epsg / making an assumption. The exception kind of gives the info that you need.

Reply:
Also get rid of the extra dots in the error message.